### PR TITLE
Homework 2

### PR DESCRIPTION
**Pull request by VShutkina**
Homework 2 added

### CODE REVIEW

File: src/test/java/hw2/ex1/FirstExerciseTest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        String actualLoggedUser = driver.findElement(By.id("user-name")).getText();
+        assertEquals(actualLoggedUser, "PITER CHAILOVSKII");
+
+        // 5. Assert Browser title
+        assertEquals(driver.getTitle(), "Home Page");
+
+        // 6. Assert that there are 4 items on the header section
+        // are displayed and they have proper texts
+        List<String> expectedHeaderListItems = Arrays.asList("HOME", "CONTACT FORM", "SERVICE", "METALS & COLORS");
+        List<WebElement> headerListItems = driver.findElements(By.xpath("//ul[@class='uui-navigation nav navbar-nav m-l8']/li"));
```
Comment (khda91) [inline src/test/java/hw2/ex1/FirstExerciseTest.java]: Why do you decide specify all classes in the locator?

File: src/test/java/hw2/ex2/SecondExerciseTest.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        WebElement serviceDatesSubCategory =
+                driver.findElement(By.xpath("//ul[@class='dropdown-menu']//li//a[contains(.,'Dates')]"));
+        WebElement serviceComplTableSubCategory =
+                driver.findElement(By.xpath("//ul[@class='dropdown-menu']//li//a[contains(.,'Complex Table')]"));
+        WebElement serviceSimplTableSubCategory =
+                driver.findElement(By.xpath("//ul[@class='dropdown-menu']//li//a[contains(.,'Simple Table')]"));
+        WebElement serviceTableWithPagesSubCategory =
+                driver.findElement(By.xpath("//ul[@class='dropdown-menu']//li//a[contains(.,'Table with pages')]"));
+        WebElement serviceDiffElemSubCategory =
+                driver.findElement(By.xpath("//ul[@class='dropdown-menu']//li//a[contains(.,'Different elements')]"));
```
Comment (khda91) [inline src/test/java/hw2/ex2/SecondExerciseTest.java]: Is it possible collapse these elements to the loop?

File: src/test/java/hw2/ex2/SecondExerciseTest.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        WebElement serviceDatesSubCategoryLeftPanel = driver.findElement(
+                By.xpath("//ul[@class='sub']//li[contains(.,'Dates')]"));
+        WebElement serviceComplTableSubCategoryLeftPanel = driver.findElement(
+                By.xpath("//ul[@class='sub']//li[contains(.,'Complex Table')]"));
+        WebElement serviceSimplTableSubCategoryLeftPanel = driver.findElement(
+                By.xpath("//ul[@class='sub']//li[contains(.,'Simple Table')]"));
+        WebElement serviceTableWithPagesSubCategoryLeftPanel = driver.findElement(
+                By.xpath("//ul[@class='sub']//li[contains(.,'Table with pages')]"));
+        WebElement serviceDiffElemSubCategoryLeftPanel = driver.findElement(
+                By.xpath("//ul[@class='sub']//li[contains(.,'Different elements')]"));
```
Comment (khda91) [inline src/test/java/hw2/ex2/SecondExerciseTest.java]: Is it possible collapse these elements to the loop and use `List<WebElement>`

File: src/test/java/hw2/ex2/SecondExerciseTest.java [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        WebElement dropdown = driver.findElement(By.cssSelector(".colors"));
+        softAssert.assertTrue(dropdown.isDisplayed());
+        WebElement checkboxWater = driver.findElement(By.xpath("//label[@class='label-checkbox' and contains(.,'Water')]"));
+        softAssert.assertTrue(checkboxWater.isDisplayed());
+        WebElement checkboxEarth = driver.findElement(By.xpath("//label[@class='label-checkbox' and contains(.,'Earth')]"));
+        softAssert.assertTrue(checkboxEarth.isDisplayed());
+        WebElement checkboxWind = driver.findElement(By.xpath("//label[@class='label-checkbox' and contains(.,'Wind')]"));
+        softAssert.assertTrue(checkboxWind.isDisplayed());
+        WebElement checkboxFire = driver.findElement(By.xpath("//label[@class='label-checkbox' and contains(.,'Fire')]"));
+        softAssert.assertTrue(checkboxFire.isDisplayed());
```
Comment (khda91) [inline src/test/java/hw2/ex2/SecondExerciseTest.java]: These elements could be get as `List<WebElement>`

File: src/test/java/hw2/ex2/SecondExerciseTest.java [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        WebElement radioGold = driver.findElement(By.xpath("//label[@class='label-radio' and contains(.,'Gold')]"));
+        softAssert.assertTrue(radioGold.isDisplayed());
+        WebElement radioSilver = driver.findElement(By.xpath("//label[@class='label-radio' and contains(.,'Silver')]"));
+        softAssert.assertTrue(radioSilver.isDisplayed());
+        WebElement radioBronze = driver.findElement(By.xpath("//label[@class='label-radio' and contains(.,'Bronze')]"));
+        softAssert.assertTrue(radioBronze.isDisplayed());
+        WebElement radioSelen = driver.findElement(By.xpath("//label[@class='label-radio' and contains(.,'Selen')]"));
+        softAssert.assertTrue(radioSelen.isDisplayed());
```
Comment (khda91) [inline src/test/java/hw2/ex2/SecondExerciseTest.java]: These elements could be get as `List<WebElement>`

File: src/test/java/hw2/ex2/SecondExerciseTest.java [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+        // 18. Assert that for each checkbox there is an individual log row
+        // and value is corresponded to the status of checkbox.
+        checkboxWater.click();
+        WebElement falseLogCheckboxWater = driver.findElement(By.
+                xpath("//ul[@class='panel-body-list logs']//li[contains(.,'Water')]"));
+        assertTrue(falseLogCheckboxWater.getText().contains("Water: condition changed to false"));
+        checkboxWind.click();
+        WebElement falseLogCheckboxWind = driver.findElement(By.
+                xpath("//ul[@class='panel-body-list logs']//li[contains(.,'Wind')]"));
+        assertTrue(falseLogCheckboxWind.getText().contains("Wind: condition changed to false"));
```
Comment (khda91) [inline src/test/java/hw2/ex2/SecondExerciseTest.java]: Why do you use `Hard Assert` here

File: src/test/java/hw2/ex2/SecondExerciseTest.java [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
```
Comment (khda91) [inline src/test/java/hw2/ex2/SecondExerciseTest.java]: I think you missed one important assert here for the `soft assertions`

**Review by khda91 — COMMENTED** [review]
Please resolve comments

File: src/test/java/hw2/ex1/FirstExerciseTest.java (hunk 1 above)
Comment (VShutkina) [reply src/test/java/hw2/ex1/FirstExerciseTest.java]: Necessary elements are found only with using all classes

File: src/test/java/hw2/ex2/SecondExerciseTest.java (hunk 7 above)
Comment (VShutkina) [reply src/test/java/hw2/ex2/SecondExerciseTest.java]: fixed

File: src/test/java/hw2/ex2/SecondExerciseTest.java (hunk 6 above)
Comment (VShutkina) [reply src/test/java/hw2/ex2/SecondExerciseTest.java]: changed to soft assertions

File: src/test/java/hw2/ex2/SecondExerciseTest.java (hunk 2 above)
Comment (VShutkina) [reply src/test/java/hw2/ex2/SecondExerciseTest.java]: fixed

File: src/test/java/hw2/ex2/SecondExerciseTest.java (hunk 3 above)
Comment (VShutkina) [reply src/test/java/hw2/ex2/SecondExerciseTest.java]: fixed

File: src/test/java/hw2/ex2/SecondExerciseTest.java (hunk 5 above)
Comment (VShutkina) [reply src/test/java/hw2/ex2/SecondExerciseTest.java]: fixed

File: src/test/java/hw2/ex2/SecondExerciseTest.java (hunk 4 above)
Comment (VShutkina) [reply src/test/java/hw2/ex2/SecondExerciseTest.java]: fixed

File: src/test/java/hw2/ex1/FirstExerciseTest.java (hunk 1 above)
Comment (VShutkina) [reply src/test/java/hw2/ex1/FirstExerciseTest.java]: Locator with contain is used now

**Review by khda91 — APPROVED** [review]
Approved. could be merged to the master branch

**Pull request merged**